src/main/java/ch/njol/skript/lang/KeyProviderExpression.java

```diff
@@ -125,4 +145,30 @@ default boolean areKeysRecommended() {
 		return true;
 	}
 
+	/**
+	 * Zips the given values and keys into a map entry array.
+	 *
+	 * @param values the values to zip
+	 * @param keys the keys to zip with the values, or null to use numerical indices (1, 2, 3, ..., n)
+	 * @return an array of map entries, where each entry is a key-value pair
+	 * @param <T> the type of the values
+	 * @throws IllegalArgumentException if the keys are present and the lengths of values and keys do not match
+	 */
+	static <T> Map.Entry<String, T> @NotNull [] zip(@NotNull T @NotNull [] values, @NotNull String @Nullable [] keys) {
```

Comment:
Any reason Map.Entry was chosen over making a record? Standardization?

Reply:
There isn't really a reason. do you think I should create a record that's used everywhere rather than a map entry?

Reply:
I think it would be preferable, since this isn't necessarily coming from a context of a Map